test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,526 @@
+            MiltonTypes.IporSwapIndicator({
+                iporIndexValue: TestConstants.PERCENTAGE_3_18DEC, // iporIndexValue
+                ibtPrice: 1 * TestConstants.D18, // ibtPrice
+                ibtQuantity: 99670098970308907327800, // ibtQuantity
+                fixedInterestRate: TestConstants.PERCENTAGE_2_18DEC // fixedInterestRate, 2%
+            }) // indicator
+        );
+        miltonUsdt.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_6DEC, // totalAmount
+            TestConstants.PERCENTAGE_1_18DEC, // acceptableFixedInterestRate, 1%
+            TestConstants.LEVERAGE_18DEC // leverage
+        );
+    }
+
+    function testShouldEmitEventWhenClosePayFixedSwap18Decimals() public {
+        // given
+        MockTestnetToken daiMockedToken = getTokenDai();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(daiMockedToken), 0);
+        IpToken ipTokenDai = getIpTokenDai(address(daiMockedToken));
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        ItfMiltonDai miltonDai = getItfMiltonDai(
+            address(daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        ItfJosephDai josephDai = getItfJosephDai(
+            address(daiMockedToken),
+            address(ipTokenDai),
+            address(miltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, daiMockedToken, address(josephDai), address(miltonDai));
+        prepareMilton(miltonDai, address(josephDai), address(stanleyDai));
+        prepareJoseph(josephDai);
+        prepareIpToken(ipTokenDai, address(josephDai));
+        // when
+        _miltonSpreadModel.setCalculateQuotePayFixed(TestConstants.PERCENTAGE_6_18DEC); // 6%
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_5_18DEC, block.timestamp); // 5%, PERCENTAGE_5_18DEC
+        vm.prank(_liquidityProvider);
+        josephDai.itfProvideLiquidity(TestConstants.USD_28_000_18DEC, block.timestamp); // TestConstants.USD_28_000_18DEC
+        vm.prank(_userTwo);
+        miltonDai.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_18DEC, // totalAmount
+            TestConstants.PERCENTAGE_6_18DEC, // acceptableFixedInterestRate, 6%
+            TestConstants.LEVERAGE_18DEC // leverage
+        );
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_160_18DEC, block.timestamp); // PERCENTAGE_160_18DEC
+        vm.prank(_userTwo);
+        vm.expectEmit(true, true, false, false);
```

Comment:
Why do you use vm.expectEmit(true, true, false, false) and not vm.expectEmit(true, true, true, true)